### ISSUE TITLE
Feature request: disable write virtual files when git working tree not clean

### ISSUE THREAD

**Issue by xiaoxiangmoe** [vuejs/language-tools#3205]

**Comment by so1ve** [comment]
@xiaoxiangmoe Shall we add an option for it?

**Comment by xiaoxiangmoe** [comment]
This is a feature request proposed by @johnsoncodehk

The background is:
write virtual files may make our project code dirty, indistinguishable from virtual file and source code.

The main requirements are:
If it is detected that this is a git repo and the working tree is not clean, write virtual files will prompt you that you need to complete git commit before you can write virtual files. If this is not a git repo, no restrictions are imposed.

Since this is a small improvement, there is no need to add options. (And there are very few people who can use write virtual files)

cc @so1ve

**Comment by so1ve** [comment]
I guess we need to implement this feature in volar framework...?

**Comment by johnsoncodehk** [comment]
Yes, it should implement at https://github.com/volarjs/volar.js/blob/master/packages/vscode/src/features/writeVirtualFiles.ts.

**Comment by johnsoncodehk** [comment]
Close as the "Write Virtual Files" command has been removed.
